### PR TITLE
feat(okx): watchTickers, watchMyTrades, createOrderWs, cancelOrderWs,… [ci deploy]

### PR DESCRIPTION
**Pull request by pcriadoperez**
- Allow all `watchTicker` channels
- Add `watchTickers`
- Add `watchMyTrades` from orders channel
- Add `createOrderWs`, `editOrderWs`, `cancelOrderWs`, `cancelOrdersWs`, `cancelAllOrders`

### CODE REVIEW

**Comment by pcriadoperez** [comment]
### watchTicker
```
okx.watchTicker(BTC/USDT)
{'ask': 31007.2,
 'askVolume': 3.367e-05,
 'average': 30674.75,
 'baseVolume': 12593106.08835147,
 'bid': 30983.0,
 'bidVolume': 19.17235638,
 'change': 616.5,
 'close': 30983.0,
 'datetime': '2023-07-14T00:44:47.014Z',
 'high': 154485.3,
 'info': {'askPx': '31007.2',
          'askSz': '0.00003367',
          'bidPx': '30983',
          'bidSz': '19.17235638',
          'high24h': '154485.3',
          'instId': 'BTC-USDT',
          'instType': 'SPOT',
          'last': '30983',
          'lastSz': '0.02419975',
          'low24h': '9900',
          'open24h': '30366.5',
          'sodUtc0': '30597.3',
          'sodUtc8': '30906.3',
          'ts': '1689295487014',
          'vol24h': '12593106.08835147',
          'volCcy24h': '481115905184.001969898'},
 'last': 30983.0,
 'low': 9900.0,
 'open': 30366.5,
 'percentage': 2.0301977508109266,
 'previousClose': None,
 'quoteVolume': 481115905184.00195,
 'symbol': 'BTC/USDT',
 'timestamp': 1689295487014,
 'vwap': 38204.705162376944}
 ```

 ### watchTickers
 ```javascript
 okx.watchTickers (BTC/USDT)
{
  'BTC/USDT': {
    symbol: 'BTC/USDT',
    timestamp: 1689295577411,
    datetime: '2023-07-14T00:46:17.411Z',
    high: 154485.3,
    low: 9900,
    bid: 30985,
    bidVolume: 0.46866853,
    ask: 31007.2,
    askVolume: 0.00003367,
    vwap: 38206.046487536936,
    open: 30373.9,
    close: 30985,
    last: 30985,
    previousClose: undefined,
    change: 611.1,
    percentage: 2.0119247116768015,
    average: 30679.45,
    baseVolume: 12590952.30091913,
    quoteVolume: 481050508931.2764,
    info: {
      instType: 'SPOT',
      instId: 'BTC-USDT',
      last: '30985',
      lastSz: '0.01792711',
      askPx: '31007.2',
      askSz: '0.00003367',
      bidPx: '30985',
      bidSz: '0.46866853',
      open24h: '30373.9',
      high24h: '154485.3',
      low24h: '9900',
      sodUtc0: '30597.3',
      sodUtc8: '30906.3',
      volCcy24h: '481050508931.276415848',
      vol24h: '12590952.30091913',
      ts: '1689295577411'
    }
  }
}
```

**Comment by pcriadoperez** [comment]
### createOrderWs

```python
okx.createOrderWs(BTC/USDT,market,buy,0.001)
[{'amount': None,
  'average': None,
  'clientOrderId': 'e847386590ce4dBCd344aac6d9bbf15d',
  'cost': None,
  'datetime': None,
  'fee': None,
  'fees': [],
  'filled': None,
  'id': '599886658493849600',
  'info': {'clOrdId': 'e847386590ce4dBCd344aac6d9bbf15d',
           'ordId': '599886658493849600',
           'sCode': '0',
           'sMsg': 'Order successfully placed.',
           'tag': 'e847386590ce4dBC'},
  'lastTradeTimestamp': None,
  'lastUpdateTimestamp': None,
  'postOnly': None,
  'price': None,
  'reduceOnly': False,
  'remaining': None,
  'side': None,
  'status': None,
  'stopLossPrice': None,
  'stopPrice': None,
  'symbol': None,
  'takeProfitPrice': None,
  'timeInForce': None,
  'timestamp': None,
  'trades': [],
  'triggerPrice': None,
  'type': None}]
  ```

**Comment by pcriadoperez** [comment]
### watchMyTrades
```python
okx.watchMyTrades()
[{'amount': 0.001,
  'cost': 31.0582,
  'datetime': '2023-07-14T01:00:22.336Z',
  'fee': {'cost': -1e-06, 'currency': 'BTC'},
  'fees': [{'cost': -1e-06, 'currency': 'BTC'}],
  'id': '96776773',
  'info': {'accFillSz': '0.001',
           'algoClOrdId': '',
           'algoId': '',
           'amendResult': '',
           'amendSource': '',
           'attachAlgoClOrdId': '',
           'avgPx': '31058.2',
           'cTime': '1689296422334',
           'cancelSource': '',
           'category': 'normal',
           'ccy': '',
           'clOrdId': 'e847386590ce4dBCf3ba2762bd613b95',
           'code': '0',
           'execType': 'T',
           'fee': '-0.000001',
           'feeCcy': 'BTC',
           'fillFee': '-0.000001',
           'fillFeeCcy': 'BTC',
           'fillNotionalUsd': '31.077766666',
           'fillPnl': '0',
           'fillPx': '31058.2',
           'fillSz': '0.001',
           'fillTime': '1689296422336',
           'instId': 'BTC-USDT',
           'instType': 'SPOT',
           'lever': '0',
           'msg': '',
           'notionalUsd': '',
           'ordId': '599887906693554176',
           'ordType': 'market',
           'pnl': '0',
           'posSide': '',
           'px': '',
           'quickMgnType': '',
           'rebate': '0',
           'rebateCcy': 'USDT',
           'reduceOnly': 'false',
           'reqId': '',
           'side': 'buy',
           'slOrdPx': '',
           'slTriggerPx': '',
           'slTriggerPxType': '',
           'source': '',
           'state': 'partially_filled',
           'stpId': '',
           'stpMode': '',
           'sz': '0.001',
           'tag': 'e847386590ce4dBC',
           'tdMode': 'cash',
           'tgtCcy': 'base_ccy',
           'tpOrdPx': '',
           'tpTriggerPx': '',
           'tpTriggerPxType': '',
           'tradeId': '96776773',
           'uTime': '1689296422336'},
  'order': '599887906693554176',
  'price': 31058.2,
  'side': 'buy',
  'symbol': 'BTC/USDT',
  'takerOrMaker': 'taker',
  'timestamp': 1689296422336,
  'type': 'market'}]
  ````

**Comment by pcriadoperez** [comment]
### cancelOrderWs
```python
okx.cancelOrderWs(599888151020150784,BTC/USDT)
[{'amount': None,
  'average': None,
  'clientOrderId': 'e847386590ce4dBCed8d24dfe228d330',
  'cost': None,
  'datetime': None,
  'fee': None,
  'fees': [],
  'filled': None,
  'id': '599888151020150784',
  'info': {'clOrdId': 'e847386590ce4dBCed8d24dfe228d330',
           'ordId': '599888151020150784',
           'sCode': '0',
           'sMsg': ''},
  'lastTradeTimestamp': None,
  'lastUpdateTimestamp': None,
  'postOnly': None,
  'price': None,
  'reduceOnly': False,
  'remaining': None,
  'side': None,
  'status': None,
  'stopLossPrice': None,
  'stopPrice': None,
  'symbol': None,
  'takeProfitPrice': None,
  'timeInForce': None,
  'timestamp': None,
  'trades': [],
  'triggerPrice': None,
  'type': None}]
  ```

**Comment by pcriadoperez** [comment]
### editOrderWs
```python
okx.editOrderWs(599889495730790400,BTC/USDT,limit,buy,0.001,None)
[{'amount': None,
  'average': None,
  'clientOrderId': 'e847386590ce4dBC10e9030274eaa17e',
  'cost': None,
  'datetime': None,
  'fee': None,
  'fees': [],
  'filled': None,
  'id': '599874870272434176',
  'info': {'clOrdId': 'e847386590ce4dBC10e9030274eaa17e',
           'ordId': '599874870272434176',
           'reqId': '',
           'sCode': '0',
           'sMsg': ''},
  'lastTradeTimestamp': None,
  'lastUpdateTimestamp': None,
  'postOnly': None,
  'price': None,
  'reduceOnly': False,
  'remaining': None,
  'side': None,
  'status': None,
  'stopLossPrice': None,
  'stopPrice': None,
  'symbol': None,
  'takeProfitPrice': None,
  'timeInForce': None,
  'timestamp': None,
  'trades': [],
  'triggerPrice': None,
  'type': None}]
  ```

File: ts/src/pro/okx.ts [hunk 1]
```diff
@@ -79,10 +103,34 @@ export default class okx extends okxRest {
         });
     }
 
-    async subscribe (access, channel, symbol, params = {}) {
+    async subscribeMultiple (access, channel, symbols: string[] = undefined, params = {}) {
         await this.loadMarkets ();
+        if (symbols === undefined) {
+            symbols = this.symbols;
```
Comment (carlosmiei) [inline ts/src/pro/okx.ts:109]: @pcriadoperez this will be a big subscription

File: ts/src/okx.ts [hunk 2]
```diff
@@ -2581,18 +2551,49 @@ export default class okx extends Exchange {
...
+         * @param {float} [params.stopLoss.triggerPrice] stop loss trigger price
+         * @param {float} [params.stopLoss.price] used for stop loss limit orders, not used for stop loss market price orders
+         * @param {string} [params.stopLoss.type] 'market' or 'limit' used to specify the stop loss price type
+         * @returns {object} an [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        let request = this.createOrderRequest (symbol, type, side, amount, price, params);
+        let method = this.safeString (this.options, 'createOrder', 'privatePostTradeBatchOrders');
+        if (request['trigger'] || (request['ordType'] === 'conditional') || (type === 'oco') || (type === 'move_order_stop') || (type === 'iceberg') || (type === 'twap')) {
```
Comment (carlosmiei) [inline ts/src/okx.ts:2584]: @pcriadoperez we can't do this because if the key does not exist, it breaks the python version

File: ts/src/okx.ts [hunk 3]
```diff
@@ -2581,18 +2551,49 @@ export default class okx extends Exchange {
...
             // keep the request body the same
             // submit a single order in an array to the batch order endpoint
             // because it has a lower ratelimit
-            extendedRequest = [ this.extend (request, params) ];
-        } else {
-            throw new ExchangeError (this.id + ' createOrder() this.options["createOrder"] must be either privatePostTradeBatchOrders or privatePostTradeOrder');
+            request = [ request ];
         }
-        const response = await this[method] (extendedRequest);
+        const response = await this[method] (request);
```
Comment (carlosmiei) [inline ts/src/okx.ts:2596]: @pcriadoperez I think we might have a bug here. What if the order does not arrive inside an array?

File: ts/src/okx.ts (hunk 2 above)
Comment (carlosmiei) [reply ts/src/okx.ts:2584]: also I think you meant `ordType === 'trigger'` instead of `request['trigger`]

File: ts/src/pro/okx.ts [hunk 4]
```diff
@@ -759,6 +899,328 @@ export default class okx extends okxRest {
...
+         * @returns {object} an list of [order structures]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
+         */
+        if (symbol === undefined) {
+            throw new BadRequest (this.id + ' cancelOrderWs() requires a symbol argument');
+        }
+        await this.loadMarkets ();
+        await this.authenticate ();
+        const url = this.urls['api']['ws']['private'];
+        const messageHash = this.nonce ().toString ();
+        const clientOrderId = this.safeString (params, 'clOrdId');
```
Comment (carlosmiei) [inline ts/src/pro/okx.ts:1119]: @pcriadoperez we should always support the ccxt term `clientOrderId` as well

File: ts/src/pro/okx.ts [hunk 5]
```diff
@@ -759,6 +901,327 @@ export default class okx extends okxRest {
...
+         * @param {object} [params] extra parameters specific to the okx api endpoint
+         * @returns {object} an [order structure]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
+         */
+        await this.loadMarkets ();
+        await this.authenticate ();
+        const url = this.urls['api']['ws']['private'];
+        const messageHash = this.nonce ().toString ();
+        let op = undefined;
+        [ op, params ] = this.handleOptionAndParams (params, 'editOrderWs', 'op', 'amend-order');
+        const args = this.editOrderRequest ('599874870272434176', symbol, type, side, amount, price, params);
```
Comment (carlosmiei) [inline ts/src/pro/okx.ts:1090]: @pcriadoperez I think this is a debug leftover 😅

**Comment by carlosmiei** [comment]
LGTM ✅ 

```
 p okx watchTicker "BTC/USDT:USDT"
Python v3.10.9
CCXT v4.0.18
okx.watchTicker(BTC/USDT:USDT)
{'ask': 31206.4,
 'askVolume': 352.0,
 'average': 30903.2,
 'baseVolume': 24120722.0,
 'bid': 31206.3,
 'bidVolume': 1057.0,
 'change': 606.4,
 'close': 31206.4,
 'datetime': '2023-07-14T09:42:18.425Z',
 'high': 31855.4,
 'info': {'askPx': '31206.4',
          'askSz': '352',
          'bidPx': '31206.3',
          'bidSz': '1057',
          'high24h': '31855.4',
          'instId': 'BTC-USDT-SWAP',
          'instType': 'SWAP',
          'last': '31206.4',
          'lastSz': '23',
          'low24h': '30458.4',
          'open24h': '30600',
          'sodUtc0': '31457.5',
          'sodUtc8': '30919.1',
          'ts': '1689327738425',
          'vol24h': '24120722',
          'volCcy24h': '241207.22'},
 'last': 31206.4,
 'low': 30458.4,
 'open': 30600.0,
 'percentage': 1.9816993464052286,
 'previousClose': None,
 'quoteVolume': None,
 'symbol': 'BTC/USDT:USDT',
 'timestamp': 1689327738425,
 'vwap': None}
```

```
 p okx watchTickers '["LTC/USDT", "BTC/USDT:USDT"]'
Python v3.10.9
CCXT v4.0.18
okx.watchTickers(['LTC/USDT', 'BTC/USDT:USDT'])
{'LTC/USDT': {'ask': 100.57,
              'askVolume': 20.268104,
              'average': 100.805,
              'baseVolume': 861663.79039,
              'bid': 100.56,
              'bidVolume': 48.786393,
              'change': -0.47,
              'close': 100.57,
              'datetime': '2023-07-14T09:54:46.025Z',
              'high': 104.45,
              'info': {'askPx': '100.57',
                       'askSz': '20.268104',
                       'bidPx': '100.56',
                       'bidSz': '48.786393',
                       'high24h': '104.45',
                       'instId': 'LTC-USDT',
                       'instType': 'SPOT',
                       'last': '100.57',
                       'lastSz': '3.040368',
                       'low24h': '99.09',
                       'open24h': '101.04',
                       'sodUtc0': '101.87',
                       'sodUtc8': '102.57',
                       'ts': '1689328486025',
                       'vol24h': '861663.79039',
                       'volCcy24h': '87660780.65838038'},
              'last': 100.57,
              'low': 99.09,
              'open': 101.04,
              'percentage': -0.4651623119556611,
              'previousClose': None,
              'quoteVolume': 87660780.65838037,
              'symbol': 'LTC/USDT',
              'timestamp': 1689328486025,
              'vwap': 101.73432101481716}}
{'BTC/USDT:USDT': {'ask': 31218.6,
                   'askVolume': 1121.0,
                   'average': 30887.95,
                   'baseVolume': 24012219.0,
                   'bid': 31218.5,
                   'bidVolume': 280.0,
                   'change': 661.5,
                   'close': 31218.7,
                   'datetime': '2023-07-14T09:54:48.191Z',
                   'high': 31855.4,
                   'info': {'askPx': '31218.6',
                            'askSz': '1121',
                            'bidPx': '31218.5',
                            'bidSz': '280',
                            'high24h': '31855.4',
                            'instId': 'BTC-USDT-SWAP',
                            'instType': 'SWAP',
                            'last': '31218.7',
                            'lastSz': '3',
                            'low24h': '30458.4',
                            'open24h': '30557.2',
                            'sodUtc0': '31457.5',
                            'sodUtc8': '30919.1',
                            'ts': '1689328488191',
                            'vol24h': '24012219',
                            'volCcy24h': '240122.19'},
                   'last': 31218.7,
                   'low': 30458.4,
                   'open': 30557.2,
                   'percentage': 2.1647925857081147,
                   'previousClose': None,
                   'quoteVolume': None,
                   'symbol': 'BTC/USDT:USDT',
                   'timestamp': 1689328488191,
                   'vwap': None}}
{'BTC/USDT:USDT': {'ask': 31218.6,
                   'askVolume': 1109.0,
                   'average': 30887.95,
                   'baseVolume': 24012219.0,
                   'bid': 31218.5,
                   'bidVolume': 280.0,
                   'change': 661.5,
                   'close': 31218.7,
                   'datetime': '2023-07-14T09:54:48.403Z',
                   'high': 31855.4,
                   'info': {'askPx': '31218.6',
                            'askSz': '1109',
                            'bidPx': '31218.5',
                            'bidSz': '280',
                            'high24h': '31855.4',
                            'instId': 'BTC-USDT-SWAP',
                            'instType': 'SWAP',
                            'last': '31218.7',
                            'lastSz': '3',
                            'low24h': '30458.4',
                            'open24h': '30557.2',
                            'sodUtc0': '31457.5',
                            'sodUtc8': '30919.1',
                            'ts': '1689328488403',
                            'vol24h': '24012219',
                            'volCcy24h': '240122.19'},
                   'last': 31218.7,
                   'low': 30458.4,
                   'open': 30557.2,
                   'percentage': 2.1647925857081147,
                   'previousClose': None,
                   'quoteVolume': None,
                   'symbol': 'BTC/USDT:USDT',
                   'timestamp': 1689328488403,
                   'vwap': None}}
{'BTC/USDT:USDT': {'ask': 31218.6,
                   'askVolume': 1120.0,
                   'average': 30887.95,
                   'baseVolume': 24012219.0,
                   'bid': 31218.5,
                   'bidVolume': 280.0,
                   'change': 661.5,
                   'close': 31218.7,
                   'datetime': '2023-07-14T09:54:48.513Z',
                   'high': 31855.4,
                   'info': {'askPx': '31218.6',
                            'askSz': '1120',
                            'bidPx': '31218.5',
                            'bidSz': '280',
                            'high24h': '31855.4',
                            'instId': 'BTC-USDT-SWAP',
                            'instType': 'SWAP',
                            'last': '31218.7',
                            'lastSz': '3',
                            'low24h': '30458.4',
                            'open24h': '30557.2',
                            'sodUtc0': '31457.5',
                            'sodUtc8': '30919.1',
                            'ts': '1689328488513',
                            'vol24h': '24012219',
                            'volCcy24h': '240122.19'},
                   'last': 31218.7,
                   'low': 30458.4,
                   'open': 30557.2,
                   'percentage': 2.1647925857081147,
                   'previousClose': None,
                   'quoteVolume': None,
                   'symbol': 'BTC/USDT:USDT',
                   'timestamp': 1689328488513,
                   'vwap': None}}
```

```
✗ p okx createOrder "LTC/USDT" limit buy 2 50  --sandbox          
Python v3.10.9
CCXT v4.0.18
okx.createOrder(LTC/USDT,limit,buy,2,50)
{'amount': None,
 'average': None,
 'clientOrderId': 'e847386590ce4dBC8c74d50e5b0f5c9',
 'cost': None,
 'datetime': None,
 'fee': None,
 'fees': [],
 'filled': None,
 'id': '600030243411288066',
 'info': {'clOrdId': 'e847386590ce4dBC8c74d50e5b0f5c9',
          'ordId': '600030243411288066',
          'sCode': '0',
          'sMsg': 'Order placed',
          'tag': 'e847386590ce4dBC'},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': False,
 'remaining': None,
 'side': 'buy',
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'LTC/USDT',
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': 'limit'}
```

```
 p okx createOrder "LTC/USDT" market  buy 0.1  --sandbox
Python v3.10.9
CCXT v4.0.18
okx.createOrder(LTC/USDT,market,buy,0.1)
{'amount': None,
 'average': None,
 'clientOrderId': 'e847386590ce4dBCc20fc63eaf8d20a0',
 'cost': None,
 'datetime': None,
 'fee': None,
 'fees': [],
 'filled': None,
 'id': '600030316966797312',
 'info': {'clOrdId': 'e847386590ce4dBCc20fc63eaf8d20a0',
          'ordId': '600030316966797312',
          'sCode': '0',
          'sMsg': 'Order placed',
          'tag': 'e847386590ce4dBC'},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': False,
 'remaining': None,
 'side': 'buy',
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'LTC/USDT',
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': 'market'}
```

```
 p okx createOrder "LTC/USDT" limit buy 2 50 '{"stopPrice":51}'  --sandbox
Python v3.10.9
CCXT v4.0.18
okx.createOrder(LTC/USDT,limit,buy,2,50,{'stopPrice': 51})
{'amount': None,
 'average': None,
 'clientOrderId': 'e847386590ce4dBCe44fc39b4403d8a',
 'cost': None,
 'datetime': None,
 'fee': None,
 'fees': [],
 'filled': None,
 'id': '600030413666648064',
 'info': {'algoClOrdId': 'e847386590ce4dBCe44fc39b4403d8a',
          'algoId': '600030413666648064',
          'clOrdId': 'e847386590ce4dBCe44fc39b4403d8a',
          'closeFraction': '',
          'sCode': '0',
          'sMsg': ''},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': False,
 'remaining': None,
 'side': 'buy',
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'LTC/USDT',
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': 'limit'}
```

```
 p okx cancelOrderWs "600030243411288066" "LTC/USDT" --sandbox
Python v3.10.9
CCXT v4.0.18
okx.cancelOrderWs(600030243411288066,LTC/USDT)
[{'amount': None,
  'average': None,
  'clientOrderId': 'e847386590ce4dBC8c74d50e5b0f5c9',
  'cost': None,
  'datetime': None,
  'fee': None,
  'fees': [],
  'filled': None,
  'id': '600030243411288066',
  'info': {'clOrdId': 'e847386590ce4dBC8c74d50e5b0f5c9',
           'ordId': '600030243411288066',
           'sCode': '0',
           'sMsg': ''},
  'lastTradeTimestamp': None,
  'lastUpdateTimestamp': None,
  'postOnly': None,
  'price': None,
  'reduceOnly': False,
  'remaining': None,
  'side': None,
  'status': None,
  'stopLossPrice': None,
  'stopPrice': None,
  'symbol': None,
  'takeProfitPrice': None,
  'timeInForce': None,
  'timestamp': None,
  'trades': [],
  'triggerPrice': None,
  'type': None}]
```

```
p okx createOrderWs "LTC/USDT" market  buy 0.1  --sandbox
Python v3.10.9
CCXT v4.0.18
okx.createOrderWs(LTC/USDT,market,buy,0.1)
[{'amount': None,
  'average': None,
  'clientOrderId': 'e847386590ce4dBC9d8e1421ca6cbc80',
  'cost': None,
  'datetime': None,
  'fee': None,
  'fees': [],
  'filled': None,
  'id': '600031913125961728',
  'info': {'clOrdId': 'e847386590ce4dBC9d8e1421ca6cbc80',
           'ordId': '600031913125961728',
           'sCode': '0',
           'sMsg': 'Order successfully placed.',
           'tag': 'e847386590ce4dBC'},
  'lastTradeTimestamp': None,
  'lastUpdateTimestamp': None,
  'postOnly': None,
  'price': None,
  'reduceOnly': False,
  'remaining': None,
  'side': None,
  'status': None,
  'stopLossPrice': None,
  'stopPrice': None,
  'symbol': None,
  'takeProfitPrice': None,
  'timeInForce': None,
  'timestamp': None,
  'trades': [],
  'triggerPrice': None,
  'type': None}]
```

```
 p okx createOrderWs "LTC/USDT:USDT" market buy 1 None '{"posSide":"long"}'  --sandbox
Python v3.10.9
CCXT v4.0.18
okx.createOrderWs(LTC/USDT:USDT,market,buy,1,None,{'posSide': 'long'})
[{'amount': None,
  'average': None,
  'clientOrderId': 'e847386590ce4dBCc58a77ac8512ae19',
  'cost': None,
  'datetime': None,
  'fee': None,
  'fees': [],
  'filled': None,
  'id': '600032134891397120',
  'info': {'clOrdId': 'e847386590ce4dBCc58a77ac8512ae19',
           'ordId': '600032134891397120',
           'sCode': '0',
           'sMsg': 'Order successfully placed.',
           'tag': 'e847386590ce4dBC'},
  'lastTradeTimestamp': None,
  'lastUpdateTimestamp': None,
  'postOnly': None,
  'price': None,
  'reduceOnly': False,
  'remaining': None,
  'side': None,
  'status': None,
  'stopLossPrice': None,
  'stopPrice': None,
  'symbol': None,
  'takeProfitPrice': None,
  'timeInForce': None,
  'timestamp': None,
  'trades': [],
  'triggerPrice': None,
  'type': None}]
```

```
 p okx watchMyTrades --sandbox                    
Python v3.10.9
CCXT v4.0.22
okx.watchMyTrades()
[{'amount': 1.0,
  'cost': 100.12,
  'datetime': '2023-07-14T10:41:35.309Z',
  'fee': {'cost': -5.006e-05, 'currency': 'USDT'},
  'fees': [{'cost': -5.006e-05, 'currency': 'USDT'}],
  'id': '23000629',
  'info': {'accFillSz': '1',
           'algoClOrdId': '',
           'algoId': '',
           'amendResult': '',
           'amendSource': '',
           'attachAlgoClOrdId': '',
           'avgPx': '100.12',
           'cTime': '1689331295308',
           'cancelSource': '',
           'category': 'normal',
           'ccy': '',
           'clOrdId': 'e847386590ce4dBCebd78c8d2840acbc',
           'code': '0',
           'execType': 'T',
           'fee': '-0.00005006',
           'feeCcy': 'USDT',
           'fillFee': '-0.00005006',
           'fillFeeCcy': 'USDT',
           'fillNotionalUsd': '0.1001830756',
           'fillPnl': '0',
           'fillPx': '100.12',
           'fillSz': '1',
           'fillTime': '1689331295309',
           'instId': 'LTC-USDT-SWAP',
           'instType': 'SWAP',
           'lever': '3',
           'msg': '',
           'notionalUsd': '0.1022243608',
           'ordId': '600034174547877888',
           'ordType': 'market',
           'pnl': '0',
           'posSide': 'long',
           'px': '',
           'quickMgnType': '',
           'rebate': '0',
           'rebateCcy': 'USDT',
           'reduceOnly': 'false',
           'reqId': '',
           'side': 'buy',
           'slOrdPx': '',
           'slTriggerPx': '',
           'slTriggerPxType': '',
           'source': '',
           'state': 'filled',
           'stpId': '',
           'stpMode': '',
           'sz': '1',
           'tag': 'e847386590ce4dBC',
           'tdMode': 'cross',
           'tgtCcy': '',
           'tpOrdPx': '',
           'tpTriggerPx': '',
           'tpTriggerPxType': '',
           'tradeId': '23000629',
           'uTime': '1689331295310'},
  'order': '600034174547877888',
  'price': 100.12,
  'side': 'buy',
  'symbol': 'LTC/USDT:USDT',
  'takerOrMaker': 'taker',
  'timestamp': 1689331295309,
  'type': 'market'}]
```

```
n okx createOrderWs "LTC/USDT:USDT" limit buy 1 50 '{"posSide":"long"}'  --sandbox
2023-07-14T10:56:45.845Z
Node.js: v18.14.0
CCXT v4.0.22
okx.createOrderWs (LTC/USDT:USDT, limit, buy, 1, 50, [object Object])
2023-07-14T10:56:47.968Z iteration 0 passed in 1431 ms

                id |                    clientOrderId | timestamp | datetime | lastTradeTimestamp | lastUpdateTimestamp | symbol | type | timeInForce | postOnly | side | price | stopLossPrice | takeProfitPrice | stopPrice | triggerPrice | average | cost | amount | filled | remaining | status | fee | trades | reduceOnly | fees
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
600038002475229184 | e847386590ce4dBCb1d38c9a149288e9 |           |          |                    |                     |        |      |             |          |      |       |               |                 |           |              |         |      |        |        |           |        |     |     [] |      false |   []
1 objects
2023-07-14T10:56:47.968Z iteration 1 passed in 1431 ms
```

```
 p okx editOrder "600038002475229184" "LTC/USDT:USDT" limit buy 01 51 --sandbox
Python v3.10.9
CCXT v4.0.22
okx.editOrder(600038002475229184,LTC/USDT:USDT,limit,buy,1,51)
{'amount': None,
 'average': None,
 'clientOrderId': 'e847386590ce4dBCb1d38c9a149288e9',
 'cost': None,
 'datetime': None,
 'fee': None,
 'fees': [],
 'filled': None,
 'id': '600038002475229184',
 'info': {'clOrdId': 'e847386590ce4dBCb1d38c9a149288e9',
          'ordId': '600038002475229184',
          'reqId': '',
          'sCode': '0',
          'sMsg': ''},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': False,
 'remaining': None,
 'side': 'buy',
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'LTC/USDT:USDT',
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': 'limit'}
```

```
 p okx editOrderWs "600038002475229184" "LTC/USDT:USDT" limit buy 01 52 --sandbox
Python v3.10.9
CCXT v4.0.22
okx.editOrderWs(600038002475229184,LTC/USDT:USDT,limit,buy,1,52)
[{'amount': None,
  'average': None,
  'clientOrderId': 'e847386590ce4dBCb1d38c9a149288e9',
  'cost': None,
  'datetime': None,
  'fee': None,
  'fees': [],
  'filled': None,
  'id': '600038002475229184',
  'info': {'clOrdId': 'e847386590ce4dBCb1d38c9a149288e9',
           'ordId': '600038002475229184',
           'reqId': '',
           'sCode': '0',
           'sMsg': ''},
  'lastTradeTimestamp': None,
  'lastUpdateTimestamp': None,
  'postOnly': None,
  'price': None,
  'reduceOnly': False,
  'remaining': None,
  'side': None,
  'status': None,
  'stopLossPrice': None,
  'stopPrice': None,
  'symbol': None,
  'takeProfitPrice': None,
  'timeInForce': None,
  'timestamp': None,
  'trades': [],
  'triggerPrice': None,
  'type': None}]
```

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**